sandbox/cgroup/tracking.go

```diff
@@ -0,0 +1,282 @@
+	//   Name string
+	//   Properties []struct{
+	//   	Name string
+	//   	Value interface{}
+	//	 }
+	// } // auxUnits describe any additional units to define.
+	type property struct {
+		Name  string
+		Value interface{}
+	}
+	type auxUnit struct {
+		Name  string
+		Props []property
+	}
+
+	// The mode string decides how the job is interacting with other systemd
+	// jobs on the system. The documentation of the systemd StartUnit() method
+	// describes the possible values and their properties:
+	//
+	// >> StartUnit() enqeues a start job, and possibly depending jobs. Takes
+	// >> the unit to activate, plus a mode string. The mode needs to be one of
+	// >> replace, fail, isolate, ignore-dependencies, ignore-requirements. If
+	// >> "replace" the call will start the unit and its dependencies, possibly
+	// >> replacing already queued jobs that conflict with this. If "fail" the
+	// >> call will start the unit and its dependencies, but will fail if this
+	// >> would change an already queued job. If "isolate" the call will start
+	// >> the unit in question and terminate all units that aren't dependencies
+	// >> of it. If "ignore-dependencies" it will start a unit but ignore all
+	// >> its dependencies. If "ignore-requirements" it will start a unit but
+	// >> only ignore the requirement dependencies. It is not recommended to
+	// >> make use of the latter two options. Returns the newly created job
+	// >> object.
+	//
+	// Here we choose "fail" to match systemd-run.
+	mode := "fail"
+	properties := []property{{"PIDs", []uint{uint(pid)}}}
+	aux := []auxUnit(nil)
+	systemd := conn.Object("org.freedesktop.systemd1", "/org/freedesktop/systemd1")
+	call := systemd.Call(
+		"org.freedesktop.systemd1.Manager.StartTransientUnit",
+		0,
+		unitName,
+		mode,
+		properties,
+		aux,
+	)
+	var job dbus.ObjectPath
+	if err := call.Store(&job); err != nil {
+		if dbusErr, ok := err.(dbus.Error); ok {
+			logger.Debugf("StartTransientUnit failed with %q: %v", dbusErr.Name, dbusErr.Body)
+			// Some specific DBus errors have distinct handling.
+			switch dbusErr.Name {
+			case "org.freedesktop.DBus.Error.NameHasNoOwner":
+				// Nothing is providing systemd bus name. This is, most likely,
+				// an Ubuntu 14.04 system with the special deputy systemd.
+				return errDBusNameHasNoOwner
+			case "org.freedesktop.DBus.Error.UnknownMethod":
+				// The DBus API is not supported on this system. This can happen on
+				// very old versions of Systemd, for instance on Ubuntu 14.04.
+				return errDBusUnknownMethod
+			case "org.freedesktop.DBus.Error.Spawn.ChildExited":
+				// We tried to socket-activate dbus-daemon or bus-activate
+				// systemd --user but it failed.
+				return errDBusSpawnChildExited
+			case "org.freedesktop.systemd1.UnitExists":
+				// Starting a scope with a name that already exists is an
+				// error. Normally this should never happen.
+				return fmt.Errorf("cannot create transient scope: scope %q clashed: %s", unitName, err)
```

Comment:
With all the randomness of uuid, should we return something like errUnitNameConflict and retry in the caller?

Reply:
Not sure, I think this is a YAGNI at least for now. I can add a comment to explore this.